pandas/core/frame.py

```diff
@@ -5838,7 +5848,15 @@ class    max    type
                         level_values, lab, allow_fill=True, fill_value=lev._na_value
                     )
 
-                new_obj.insert(0, name, level_values)
+                if allow_duplicates is None:
+                    allow_duplicates = self.flags.allows_duplicate_labels
```

Comment:
Micro optimization - can you move this outside the for loop? It's constant and doesn't need to be evaluated repeatedly.

Reply:
Done